WORKSPACE

```diff
@@ -61,6 +61,17 @@ crate_universe_dependencies(bootstrap = True)
 load("//third_party/rust/crates:crates.bzl", "crate_repositories")
 crate_repositories()
 
+load("@rules_rust//crate_universe:defs.bzl", "crates_repository")
+crates_repository(
+    name = "mdbook_deps",
+    cargo_lockfile = "@mdbook//:Cargo.lock",
+    manifests = ["@mdbook//:Cargo.toml"],
+    # @cargo_bazel_bootstrap remote created in crate_universe_dependancies
+    generator = "@cargo_bazel_bootstrap//:cargo-bazel"
+)
+load("@mdbook_deps//:defs.bzl", mdbook_deps="crate_repositories")
+mdbook_deps()
+
```

Comment:
It looks like the mdbook package contains both libraries and binaries.  We should be able to add `mdbook` to our normal //third_party/rust/Cargo.toml file and then provide a `crate.annotation` in `//third_party/rust/BUILD` to instruct bazel to make the binaries available.

We could also supply patches there and thus use the upstream crate rather than a repo from your personal github account.

```
annotations = {
    ...
    "mdbook": [crate.annotation(
        patch_args = ["-p1"],
        patches = ["@//third_party/rust/patches:mdbook-landing-page-links.patch"],
        gen_binaries = True,
    )],
}
```

Once that is in place, you'll need to regenerate the vendored BUILD files:
```
bazel run //third_party/rust:crate_index -- --repin
```

That should prevent us from having to maintain a separate `mdbook_deps` and hand-written build rules.  We can also trade importing a forked/patched repo for applying a patch to the canonical upstream repo.